crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1495,48 +1491,48 @@ impl TempSymbolProvider {
             .ok()
     }
 
-    fn into_missing_ids(self) -> BTreeSet<DebugId> {
-        self.missing_ids.into_inner()
+    fn missing_ids(self) -> Vec<DebugId> {
+        self.missing_ids.into_inner().into_iter().collect()
     }
 }
 
+#[async_trait]
 impl SymbolProvider for TempSymbolProvider {
-    fn fill_symbol(
+    async fn fill_symbol(
         &self,
-        module: &dyn Module,
-        _frame: &mut dyn minidump_processor::FrameSymbolizer,
-    ) -> Result<(), minidump_processor::FillSymbolError> {
-        // TODO(ja): Deduplicate this. Probably should use a different map key, ...
-        let debug_id = module.debug_identifier().unwrap_or_default();
-
+        module: &(dyn Module + Sync),
+        _frame: &mut (dyn FrameSymbolizer + Send),
+    ) -> Result<(), FillSymbolError> {
         // Symbolicator's CFI caches never store symbolication information. However, we could hook
         // up symbolic here to fill frame info right away. This requires a larger refactor of
         // minidump processing and the types, however.
         // TODO(ja): Check if this is OK. Shouldn't trigger skip heuristics
-        if self.files.contains_key(&debug_id) {
-            Ok(())
-        } else {
-            Err(minidump_processor::FillSymbolError {})
+        match module
+            .debug_identifier()
+            .map(|debug_id| self.files.contains_key(&debug_id))
+            .unwrap_or_default()
```

Comment:
Mixed feelings seeing this change because I think it makes it more difficult to see the main operation being acted upon here (`self.files.contains_key`), but I don't think it's a dealbreaker. Looks like logically this is the same, so this is more of a readability nit than anything else.